### PR TITLE
[PUI] Error pages

### PR DESCRIPTION
**Pull request by SchrodingersGat**
- Adds framework for displaying various error pages to the user based on API response status
- Prevents non-staff users from accessing system settings
- Prevents non-staff users from accessing admin page
- Handle 404 errors (and other errors) for detail view pages
- Closes https://github.com/inventree/InvenTree/issues/5755

### CODE REVIEW

File: src/frontend/src/pages/InstanceDetail.tsx
Comment (matmair) [inline src/frontend/src/pages/InstanceDetail.tsx:1]: Comment: Shouldn't this be in layout / components?

File: src/frontend/src/pages/InstanceDetail.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    return <LoadingOverlay />;
+  }
+
+  if (status >= 500) {
+    return <ServerError status={status} />;
+  }
+
+  if (status >= 400) {
+    return <ClientError status={status} />;
+  }
```
Comment (matmair) [inline src/frontend/src/pages/InstanceDetail.tsx:25]: Question: What is the error state if the user has viewing perms but can not do a change action? Are they immediately thrown off the page?

**Review by matmair — APPROVED** [review]
LGTM;
Site Note: It seems like the pattern of a navigationtree and details occurs a bunch of time - this could be a good candidate for a generic component in the future

File: src/frontend/src/pages/InstanceDetail.tsx (hunk 1 above)
Comment (SchrodingersGat) [reply src/frontend/src/pages/InstanceDetail.tsx]: I do not think that this update will have any issues here - the permissions checks should happen separately. Certainly will keep an eye out for it

**Pull request merged**